spec/features/invite_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+
+feature 'invitation' do
+
+  given(:tom) { Fabricate(:user, full_name: 'Tom, the cat') }
+  given(:jerrys_email) { 'jerry@mice.com' }
+  given(:jerrys_name) { 'Jerry, the mouse' }
+
+  background 'tom is signed-in and invites jerry' do
+
+    clear_emails
+    sign_in tom
+
+    visit invite_path
+    p page.body
+    fill_in :invitation_email, with: jerrys_email
+    fill_in :invitation_full_name, with: jerrys_name
+    click_button 'Send Invitation'
+    open_email jerrys_email
+  end
+
+  scenario 'jerry receives an email' do
+    expect(current_email.to).to include jerrys_email
+    expect(current_email.body).to have_content jerrys_name
+  end
+
+  scenario 'jerry can confirm the invitation' do
+    # workaround: replace the host:port
+    href = confirm_invitation_url(Invitation.last.token, host: 'localhost:3000')
```

Comment:
I don't know why but this solution doesn't work for me. Now, I debugged the Rails code just to see that the host is actually taken from ```ENV['HTTP_HOST']```.
This is where the ```ENV['HTTP_HOST']``` value is "example.com". The according method in url.rb is:
```
    def raw_host_with_port
        if forwarded = env["HTTP_X_FORWARDED_HOST"]
          forwarded.split(/,\s?/).last
        else
          env['HTTP_HOST'] || "#{env['SERVER_NAME'] || env['SERVER_ADDR']}:#{env['SERVER_PORT']}"
        end
      end
```

Google says that this environment variable is taken from ```Capybara.default_host```. Therefore I opened spec_helper.rb and set it as follows just to check if anything changes:

```
Capybara.default_host = 'bla.com'
```
Unfortunately, from now on many tests were failing because every ```visit *_path``` line has thrown "Not found: ..." error so that any subsequent lines of code, for instance ```fill_in ...``` failed because there was no page content. This is why I thought that 'bla.com' might need the http:// prefix or something else. Now it's clear that this configuration is correct but there is  no application executed under this host so that Capybara cannot visit this path.

However, I could fix it in that way that I didn't change the host or port in ```spec_helper.rb```. I just adapted the default_url_settings so that it fits the Capybara value.

```
config.action_mailer.default_url_options = { host: 'www.example.com' }
```

Now it's working but I'm not satisfied with this solution. I would like to understand which property is used for what and what is the proper solution for that.